sa/storage-authority.go

```diff
@@ -232,6 +232,50 @@ func (ssa *SQLStorageAuthority) GetLatestValidAuthorization(registrationID int64
 	return ssa.GetAuthorization(auth.ID)
 }
 
+// GetValidAuthorizations returns the latest authorization object for all
```

Comment:
Aha, now I see the logic below where you pick the later-expiring ones as you go through the list. You could achieve the same affect with an `ORDER BY expires ASC` and removing the conditional insertion into the map. We have this index on authz, which makes the `ORDER`, when combined with the rest of the query, cheap:

```
  KEY `registrationID_identifier_status_expires_authz_idx` (`registrationID`,`identifier`,`status`,`expires`),
```